sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvoker.java

```diff
@@ -56,11 +56,13 @@
   <RestrictionT> RestrictionT invokeGetInitialRestriction(InputT element);
 
   /** Invoke the {@link DoFn.GetRestrictionCoder} method on the bound {@link DoFn}. */
-  <RestrictionT> Coder<RestrictionT> invokeGetRestrictionCoder();
+  <RestrictionT> Coder<RestrictionT> invokeGetRestrictionCoder(CoderRegistry coderRegistry);
 
   /** Invoke the {@link DoFn.SplitRestriction} method on the bound {@link DoFn}. */
-  <RestrictionT> List<RestrictionT> invokeSplitRestriction(
-      InputT element, RestrictionT restriction);
+  <RestrictionT> void invokeSplitRestriction(
+      InputT element,
+      RestrictionT restriction,
+      DoFn.OutputReceiver<RestrictionT> restrictionReceiver);
```

Comment:
I believe `OutputReceiver` was described as "A placeholder for testing handling of output types during {@link DoFn} reflection." If we're going to use it for more, we should update that comment.